### ISSUE TITLE
`@export var n: Node3D` breaks when changing the type of the set value in the scene editor

### ISSUE THREAD

**Issue by mieldepoche** [godotengine/godot#82944]
### Godot version

4.1.1

### System information

linux

### Issue description

As seen in this clip, if you change the type of the node, it invalidates the export. (printing its value in that state shows `<Object#null>`.

If the new node type is compatible, it should keep the value clean.
If the new node type is incompatible, it should display a warning somehow (maybe a node configuration warning?).

https://github.com/godotengine/godot/assets/71937306/7f58c879-74dd-41b2-b387-c82a45824758

### Steps to reproduce

```gdscript
extends Node3D

@export var n: Node3D

func _ready() -> void:
    print(n)
```
fill out that variable in a scene and change the node type.

### Minimal reproduction project

any

**Comment by jackwilsdon** [comment]
It looks like this is a duplicate of #78299, which should be fixed in 4.2 by the looks (fixed in #78300).

**Comment by Calinou** [comment]
Thanks for the report! Consolidating in https://github.com/godotengine/godot/issues/78299, which is fixed in 4.2.dev6.
